v2/internal/genericarmclient/generic_client.go

```diff
@@ -262,7 +262,12 @@ func (client *GenericClient) deleteByID(ctx context.Context, resourceID string,
 	resourceType := metrics.GetTypeFromResourceID(resourceID)
 
 	client.metrics.RecordAzureRequestsTime(resourceType, time.Since(requestStartTime), metrics.HttpDelete)
-	client.metrics.RecordAzureRequestsTotal(resourceType, resp.StatusCode, metrics.HttpDelete)
+
+	statusCode := 0
```

Comment:
Is zero the best choice here? Actually, doesn't a lack of a response mean there was no request at all? Wouldn't it be better to skip adding it to the metrics? Feels like adding non-requests to the list of requests would be a bit like this:

How many times did you go to the store today?
I didn't go 5 times.

Reply:
I did copy this logic from the method `createOrUpdateByID`.. Although I also felt this is not the best possible solution. I'll do a small PR on changing this metric+documentation totally. So rather than having `total` requests, I'm planning to make it `successful` requests which would make more sense and we can avoid this error.